libs/utils/trace.py

```diff
@@ -429,11 +429,11 @@ def getTasks(self, dataframe=None,
         :param pid_key: The name of the dataframe columns containing task PIDs
         :type pid_key: str
         """
-        if dataframe is None:
-            return self.tasks
-        df = dataframe
         if task_names is None:
             task_names = self.tasks.keys()
+        if dataframe is None:
+            return {k: v for k, v in  self.tasks.iteritems() if k in task_names}
```

Comment:
We should fix the docstring for task_names to say "(by default all tasks in the trace)"